src/java.base/linux/classes/sun/nio/ch/EPollSelectorImpl.java

```diff
@@ -110,7 +111,10 @@ private void initFDs() throws IOException {
 
         try {
             this.eventfd = new EventFD();
-            IOUtil.configureBlocking(IOUtil.newFD(eventfd.efd()), false);
+            FileDescriptor fd = IOUtil.newFD(eventfd.efd());
+            // This FileDescriptor is a one-time use, the actual FD will be closed from EventFD
+            fd.markClosedByNIO();
+            IOUtil.configureBlocking(fd, false);
```

Comment:
The eventfd was actually closed on line 149, so it's incorrect to mark it is closed here. I think everything was fine before this chunk

Reply:
The native descriptor is closed, but the FileDescriptor instance is created and forgotten. If the GC doesn't remove it before checkpoint it may be reported as a FileDescriptor leak. I think I saw that happening.